lib/ibm/cloud/sdk/base_service.rb

```diff
@@ -25,7 +25,11 @@ def post(path, payload)
         private
 
         def url(path)
-          "#{endpoint}/#{path}"
+          if path.start_with?('https')
```

Comment:
This is a bit confusing to me, since all of our endpoints start_with 'https'. I think the idea to allow an override since the get_storage_types call has a weird path?

It would be nice to have a general way to override the endpoint (e.g. for testing) - either implemented here or in a future commit.

Reply:
Yeah :-1: for this, would prefer passing an optional endpoint to this method if we have to support this.

Would like to understand why this is needed though since if it isn't the same endpoint I'd argue it shouldn't be in this class.

Reply:
In the nearish future the `get_storage_types` call will be refactored under the common `IBM::Cloud::SDK::PowerIaas.endpoint` so I'd prefer to keep it in this class with the technical debt + a generic endpoint override.

Reply:
If this is going to be fixed "soon" we could just `JSON.parse(RestClient.get("https://#{region.sub(/-\d$/, '')}.power-iaas.cloud.ibm.com/broker/v1/storage-types", headers))` in that method rather than introduce technical debt to the base class

Reply:
:+1: I like this approach. Let's keep the ugliness contained within the method.

Reply:
I just wanted to enhance the functionality, but I understand the concern in making it stick to just one format and nothing more. Sure, I can put this unclean code :-) with a comment that it be removed as soon as the API become available.